mitodl/micromasters
# Added enrollment dialog

#### What are the relevant tickets?
Fixes #796 

#### What's this PR do?
Adds a dialog box which opens when the user clicks 'Enroll in a new program'. This dialog is used to add a new `ProgramEnrollment` for the user.

#### How should this be manually tested?
Make sure you are not enrolled in every live program. Then you should see the link in the program selector. Note that:

 - If you select the 'Enroll in a new program' option, it should not change the currently selected program displayed in the dropdown.
 - If you click 'Cancel' the dialog should close and nothing should change.
 - If you click 'Enroll' without a program selected you should get a validation warning. (The warning should disappear if you close and open the dialog again.)
 - You should see only programs you have not yet enrolled in.
 - When you select a program and click 'Enroll', it should close the dialog immediately and POST to the API. If it succeeds you should see a success message with a check mark like in the screenshot for #796. If it fails you should see an error icon with an error message. In both cases the toast will disappear after five seconds.
 - After a successful enrollment you will automatically be switched to the new program in the context.

#### Any background context you want to provide?
The click outside the dialog to close it is not supported due to complications with material-ui and react-select: https://github.com/JedWatson/react-select/issues/532
There may be some workaround for that, I haven't spent much time on it. Made an issue: #1002

Also, program page URLs are not available in the dashboard at the moment so they won't be displayed in the UI. Made an issue: #1003 

#### Screenshots (if appropriate)
![screenshot from 2016-09-09 12-39-58](https://cloud.githubusercontent.com/assets/863262/18395004/8b668360-768a-11e6-8452-ac41d04848d7.png)
